### PR TITLE
Update `cfe`, `topmodel`, and `lgar` calibratable parmeters; `ngen.cal` python < 3.9 pin `fiona<1.10.0`

### PR DESCRIPTION
**Pull request by aaraney**
## Additions

- `LgarParams`
- docstrings for `CfeParams`, `TopmodelParams`, and `LgarParams` parameters
- New `CfeParams` and `TopmodelParams` calibratable parameters

## Removals

- `CfeParams` `bb` and `multiplier` fields dropped. `bb` is now `b`; support for `multiplier` was dropped in https://github.com/NOAA-OWP/cfe/pull/64

## Changes

- `ngen.cal` on `python` `3.9` pins `fiona<1.10.0` (see notes)

## Notes:

`fiona` `v1.10.0` deprecated  `fiona.path` field (seem to have been removed?) [see](https://github.com/Toblerity/Fiona/releases/tag/1.10.0).

`geopandas` `geojson` reader uses `fiona`.

`geopandas` `v0.14` dropped support for `python<3.9`. So, `ngen.cal` installations are effectively pinned to `geopandas<=0.14`. Transitively, `ngen.cal` needs to pin `fiona<1.10.0` for compatibility.
https://geopandas.org/en/stable/docs/changelog.html#version-0-14-sep-15-2023

### CODE REVIEW

File: python/ngen_conf/src/ngen/config/cfe.py [hunk 1]
```diff
@@ -5,15 +5,65 @@
...
+    maxsmc: Optional[float]
+    """saturated soil moisture content"""
+    wltsmc: Optional[float]
+    """wilting point soil moisture content"""
+    refkdt: Optional[float]
+    """
+    Surface runoff parameter; REFKDT is a tuneable parameter that significantly
+    impacts surface infiltration and hence the partitioning of total runoff into
+    surface and subsurface runoff. Increasing REFKDT decreases surface runoff
+    """
```
Comment (hellkite500) [inline python/ngen_conf/src/ngen/config/cfe.py:29]: Probably worth noting in the docstring here that this is only used with Schaake option
```suggestion
    """
    Surface runoff parameter; REFKDT is a tuneable parameter that significantly
    impacts surface infiltration and hence the partitioning of total runoff into
    surface and subsurface runoff. Increasing REFKDT decreases surface runoff.
    Only applicable when surface_water_partitioning_scheme is Schaake.
    """
```

File: python/ngen_conf/src/ngen/config/cfe.py [hunk 2]
```diff
@@ -5,15 +5,65 @@
...
+    cgw: Optional[float]
+    """Coefficent for the groundwater equation"""
+    alpha_fc: Optional[float]
+    """field_capacity_atm_press_fraction (alpha_fc)"""
+    kn: Optional[float]
+    """Nash coefficient for subsurface lateral flow through Nash cascade"""
+    klf: Optional[float]
+    """Nash coefficient that determines the volume of lateral flow"""
+    kinf_nash_surface: Optional[float]
+    """Storage fraction per hour that moves from Nash surface reservoirs to soil"""
```
Comment (hellkite500) [inline python/ngen_conf/src/ngen/config/cfe.py:43]:
```suggestion
    """
    Storage fraction per hour that moves from Nash surface reservoirs to soil.
    Only applicable surface_runoff_scheme = NASH_CASCADE.
    """
```

File: python/ngen_conf/src/ngen/config/cfe.py [hunk 3]
```diff
@@ -5,15 +5,65 @@
...
+    retention_depth_nash_surface: Optional[float]
+    """Water retention depth threshold (only applied to the first reservoir)"""
```
Comment (hellkite500) [inline python/ngen_conf/src/ngen/config/cfe.py:45]:
```suggestion
    """
    Water retention depth threshold (only applied to the first reservoir)
    Only applicable surface_runoff_scheme = NASH_CASCADE.
    """
```

File: python/ngen_conf/src/ngen/config/cfe.py (hunk 1 above)
Comment (aaraney) [reply python/ngen_conf/src/ngen/config/cfe.py:29]: These are great suggestions! Thanks for the careful eyes! I pulled straight from `cfe`'s confirmation docs. I will make your suggested changes (tomorrow, AFK at the moment) and open a pr to cfe to improve their docs (w/ co-author of course).

**Review by hellkite500 — APPROVED** [review]

**Pull request merged**